docs/development/major-release.md

```diff
@@ -0,0 +1,45 @@
+# Major Releases
+
+## When will we push a major version?
+
+- There is no set schedule for major version releases
+- If we need to bump the **minor** (or major) version of Node.JS required to run Renovate
+- If we have a few breaking changes "piling up"
+- If we have a need to introduce something that will need wider visibility (i.e. some changes to `config:best-practices` )
+
+## Preparing
+
+- Determine what's in the next release by looking at the [Milestones view](https://github.com/renovatebot/renovate/milestones)
+
+### Draft release notes
+
+It is worthwhile drafting the release notes ahead of time, ideally on a GitHub Discussion or Issue, so you can preview what GitHub Flavoured Markdown thinks **??**.
+
+For example see [how v42 release notes were drafted](https://github.com/renovatebot/renovate/discussions/38841#discussioncomment-14770478).
+
+## Branching workflow + releasing
+
+When in the days before the release is ready to land:
+
+1. Create a temporary branch for release, eg `next-major` (from `main`)
+   - This should **not** be a protected branch (including `next`), as we will be rebasing it regularly
+1. Note issues to close as part of each PR
+1. For each planned breaking change PR, rebase them + finalise the Git commit history
+   - Including adding the `Closes #...`
+   - Force push them back to the PRs
+1. `git cherry-pick` all planned breaking PRs to that branch
+1. Create a pr from `next-major` branch to `main` (add `ci:fulltest` label)
+1. Wait for ci
+1. Rebase release branch onto main if required and goto 4.
+1. Use `git` CLI to push all commits from release branch to main
+   - **NOTE** that this is a "I'm going to `git push` and bypass branch protection requirements" rather than `git push --force`
+   - This needs to be separate commits, so they each get their own note in the CHANGELOG
+1. Wait for release
+1. Take draft release notes, and prepend them to the autogenerated release notes from Semantic Release
+
+> [!IMPORTANT]
+> The final `git push` to `main` can be performed by users with the Maintain role, which includes all Renovate maintainers.
```

Comment:
```suggestion
> The final `git push` to `main` can be performed by users with the Maintainer role, which includes all Renovate maintainers.
```

Reply:
I'm mapping this to the GitHub role `Maintain` ([more info](https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#get-repository-permissions-for-a-user)) which is separate to the Renovate maintainer team